### ISSUE TITLE
Signal notifications manage to ignore DND on various android phones

### ISSUE THREAD

**Issue by hboetes** [signalapp/Signal-Android#12708]
>> I just woke up because of this bug, please fix it.

> If you are still having this issue, please submit a new GitHub Issue with a debuglog URL. Thank you!

_Originally posted by @nicholas-signal in https://github.com/signalapp/Signal-Android/issues/8257#issuecomment-1369914895_

I tried all possible settings as proposed in the original issue to double-check I didn't have an exception for signal which would allow it to ignore DND, but still, it's ignoring DND.

Here is the link to the debug log:  https://debuglogs.org/android/6.6.3/7178b2025cfe6c18cc93cd60f56cf95618fcf21b2757584c6c8a445d97da9c53

**Comment by hboetes** [comment]
bump

**Comment by nicholas-signal** [comment]
We have been unable to reproduce this issue on our end. A few clarifying questions:

- Is it all conversations or just some?
- Do you have the contacts in question as "starred" or "priority" contacts?
- Do you have the conversations as "priority conversations" in the system settings?
- If you have other devices (or friends with other Android devices), can you confirm if it reproduces on theirs? (and if so, please share the manufacturer and device model)

**Comment by CommonLoon102** [comment]
https://github.com/signalapp/Signal-Android/issues/12900

**Comment by jerryyhom** [comment]
I have been experiencing this issue for a while but can't recall when it started.  At some point, I thought to try the Notification Profile (NP) feature, which works as expected.  However, I have a feature request: is it possible to queue notifications and send them silently when the NP disables?  I'm basically using NP as a substitute for Do Not Disturb, and notifications are dropped completely.  Ultimately, the better solution is to fix the root issue.

LG Stylo 6
Android 10
Signal v6.21.3

I will reply to @nicholas-signal questions.

> We have been unable to reproduce this issue on our end. A few clarifying questions:
> 
>     * Is it all conversations or just some?

All conversations.


>     * Do you have the contacts in question as "starred" or "priority" contacts?

No.


>     * Do you have the conversations as "priority conversations" in the system settings?

No.


>     * If you have other devices (or friends with other Android devices), can you confirm if it reproduces on theirs? (and if so, please share the manufacturer and device model)

Sorry, cannot confirm.

**Comment by marcinsmialek** [comment]
I had it on Samsung S21U, maybe even earlier, now on S23U, Android 14, EUX, Polish locale. The account data has been restored from backup, which was transfered many times during last years.

It just looks as if Signal didn't use the standard notifications channel.

**Comment by nikola197** [comment]
I have the same problem with both starred and ordinary contacts.
Phone is Samsung A52s with latest Android always.
Issue occured from OneUI 5.0, over 5.1, and latest 6.0 (Android 14)

**Comment by marcotama** [comment]
I have the same problem on my Pixel 8 Pro and I had it for a while on my Pixel 5, too.

**Comment by ElijahLynn** [comment]
I have the same issue on Pixel 7, Android 14, Build AP1A.240305.019.A1, Signal 7.2.4. The only way I can use Signal is to use a "Focus" notification profile in addition to the system DND mode. I have to manually put the notification profile on everytime I put DND mode on. Highly frustrating and I forget to do it most of the time, and then I hear a ding in DND mode and remember to turn it on. I'm happy to do a live screenshare to figure it out with a Signal engineer, ping me at elijah@elijahlynn.net if interested.

**Comment by ElijahLynn** [comment]
related:
* https://github.com/signalapp/Signal-Android/issues/8257
* https://github.com/signalapp/Signal-Android/issues/8530

**Comment by ElijahLynn** [comment]
@nicholas-signal & @cody-signal I'll pay you $50 for a 30 minute Zoom call to help debug this issue. I really want to get it fixed. Happy to even post a recording to this thread to keep all the communication opensource.

**Comment by githubuser6000** [comment]
Can confirm. I don't understand why signal consistently fails to follow settings that other similar apps have no problem following. Do not disturb does not work. Silent mode still causes the ringer to go off when a call arrives. Setting a contact to mute (doesn't matter how long) still causes the call to go through. It's getting _incredibly_ frustrating.

**Comment by greyson-signal** [comment]
@githubuser6000 I think muting may only apply to messages, but everything else you mention should work. We'll look into it.

**Comment by ElijahLynn** [comment]
Thank you @greyson-signal. I'm available for realtime debugging if your engineering team wants to pair on this. Email me at elijah@elijahlynn.net if you want to sync.

**Comment by ChimeraCoder** [comment]
I'm running into this issue as well. It just happened to me twice, about two hours apart, on a Pixel 8 Pro running Android 14. 

* I had Do Not Disturb enabled system-wide
* "Calls that can interrupt" is set to None
* "Alarms and other interruptions" is set to "Alarms" and "Media sounds" 
*  I only have one notification profile within Signal, and it is disabled 
* Signal does not have permission to access/override DND in my system settings. 

That third bullet point makes me suspect that somehow the Signal ringtone is being treated as a media sound, although I don't know how or why this would happen.

**Comment by tve** [comment]
I'm also having issues with the system wide DND: incoming calls ring loudly!
Pixel 7, Android 14, signal 7.18.2
I have no notification profiles.
Something i find suspicious is that if i turn DND on using the system drop down panel and then in signal on the chats page hit the physical volume up/down button then the volume slider is greyed but above it the icon shows "sound" and not vibrate.

**Comment by keithritter** [comment]
Same issue here.
Pixel 9, Android 14, newest Signal version. 
No notification profiles. Same setup as ChimeraRider except I have "allow repeated calls" on.
I should note that sometimes it effectively blocks the calls and sometimes it lets them through. I've even had calls silenced on the first few rings and then the last ring or two is audible. I should also note I've started getting another bug in the last couple days, which is that I often don't receive calls when DnD isn't even on. I just receive a missed call message right after. My Signal ringing capability is screwed all around. Would a factory reset be worth trying? 
I'll be monitoring this thread and hope you can make progress. For now, I'm going to have to mute all Signal notifications when I sleep at night and remember to unmute in the morning.

**Comment by keithritter** [comment]
> As a low-tech workaround, what if not having the phone in the same room where you sleep? You can even put clothes on it. Then no need to mute/unmute Signal notifications all the time.

Believe me, I wish I could. Healthier sleep patterns and all that. I sometimes have to take work calls though. I also have an elderly mother and I'd be afraid of missing an emergency call about her health.

**Comment by ChimeraCoder** [comment]
Some more information that may help:

I have run into this issue more than once, with _both_ text notifications and phone calls, as well as "missed call" messages. When I say that I have run into it with text messages, I'm referring to the ringtone for a text message going through despite Do Not Disturb being enabled, which is a separate issue from https://github.com/signalapp/Signal-Android/issues/13094. 

I recently replaced my phone (same make/model, but a fresh installation on a new physical device) and have run into the issue on both the old and the new device.


@greyson-signal I captured a debug log a few minutes after the last time that it happened. It confirms that the only Signal profile I have was not active - not sure if any other info from that would be helpful.

**Comment by ElijahLynn** [comment]
Okay, another approach...

Who wants to pledge with me to donate to the Signal foundation once this is fixed? 

I pledge $50, who wants to match me? Can we get to $500?

**Comment by githubuser6000** [comment]
Surprisingly enough, if you enable "ringer off" mode, signal does not audibly play the ringer. 

Looks like this in the volume-slider popup 
![image](https://github.com/user-attachments/assets/6cbd3763-e59b-49ee-b7f0-2dfaa5b204fa)

Looks like this in the notification / status bar 
![image](https://github.com/user-attachments/assets/a853a050-28d5-477b-b9bb-ce00345564d1)

**Comment by keithritter** [comment]
I've been turning off Signal notifications along with using Do Not Disturb to insure I'm not disturbed but can still get emergency calls from repeat callers on my regular number. 
This morning I got woken up by a Signal call even with Signal notifications turned off and Do Not Disturb on. This bug is really something. After thinking I might need to silence my phone completely or keep the phone out of my room, I thought of a temporary solution. Change the Signal ringtone to silent before bed or meetings and change the ringtone back after. Tedious, but hopefully it should work for now.

**Comment by ChimeraCoder** [comment]
I am curious for folks who have been experiencing this issue: how many of you have "Allow repeat callers" enabled? (Notifications > Do Not Disturb > People > Calls > Allow Repeat Callers).

As an experiment, I met this weekend with one of my contacts whose call had rung despite DND. When I enabled "Allow repeat callers" and called myself from their phone, it rang, even though that was the first call within the 15 minute window. When I disabled that and tried again, it did not ring.

It seems that, for some reason, Signal's call notification is registering as a repeat call from that caller, even when it is not. 

That said, I don't think this is the only bug at hand. That particular setting is scoped only to calls, but I have encountered this issue with Signal text messages as well.

**Comment by marcinsmialek** [comment]
> I am curious for folks who have been experiencing this issue: how many of you have "Allow repeat callers" enabled? (Notifications > Do Not Disturb > People > Calls > Allow Repeat Callers).
> 
> (...)

I've just checked that on my S23U. I had this feature enabled, but even after disabling it, the phone vibrated. However... when I **removed** my brother **from emergency contacts**, it didn't  vibrate when we tested, even with repeated calls notification enabled.

**Comment by CGA11** [comment]
I haven't had this problem for quite a while, but yesterday it reappeared. Message notifications from contacts break DND on my Redmi Note 12.

**Comment by mildly-useful** [comment]
It's happening to me on Android 14 with the latest version of Signal. 

Signal does not have access to override DND priority, calls that  interrupt" is "none," and "Alarms and other interruptions" is set to "Alarms and media sounds." Repeat callers is "ON"

So annoying... It seems I have to literally mute all Signal notifications. I'm sleeping soundly and Signal wakes me up despite DND being ON.

**Comment by githubuser6000** [comment]
There is a literal "play" icon in the small bubble when you're on a video call. 

> Either that.. or the Signal ringer is a "media" sound.

That's my guess too. "Ringer off" mode seems to be working to keep singal calls at bay. 

I am almost certain signal is "emulating" a call and not doing what it is supposed to.

**Comment by mildly-useful** [comment]
Even with the Ringtone set to None/Silent from the Calls options, I am still getting audible calls. What the hell? lol.

**Comment by ElijahLynn** [comment]
$100 to the Signal Foundation when this issue is fixed, that is my pledge.

**Comment by githubuser6000** [comment]
> I literally cannot silence this app.

Uninstall it lol /s 

Or wait till all of your contacts get fed up with signal and then you'll get no calls because there is no one to call /s

> $100 to the Signal Foundation when this issue is fixed, that is my pledge.

Without a roadmap from the foundation, this won't do much I guess. But hey if someone from signal can post semi-regular updates on this, I'll hop on this bandwagon.

**Comment by DanielBarie** [comment]
Also experiencing this. Three times in the course of this week. Priority contact or not doesn't matter. Phone is set to DND mode but Signal keeps ringing and vibrating. Any other app does respect settings (a normal call came in just two minutes prior which was suppressed). DND may only be circumvented by repeat callers (all calls were first time/one-off) and alarms (no media, no touch tones, no reminders, no calendar notifications, no acoustic signals for notifications may override dnd). No overrides in Signal. 
@mildly-useful suspects the signal ring tone to be of media type, this probably is not the case (rings even when media is off in dnd mode).
@ChimeraCoder seems to be on the right track (allow repeat callers is on). I need to be available for repeat callers even when in dnd mode. So switching this off is not an option. Signal really needs to respect this setting.
Phone: Pixel 6a
OS: Android 14 (Sept 2024)

**Comment by CommonLoon102** [comment]
Buy a dumb phone with and new number, give the number to those who can wake you up. They can call you via normal call and you can call them back with Signal.

**Comment by ferricoxide** [comment]
Ran into this last night: wife was up late waiting for the snowpocalypse and ended up sending me a message with video attachment, one with picture-attachement plus two text-only messages at 03:15. Got chirped awake, even though my (OnePlus 9Pro w/ Android 14) phone was in automatic DND mode. Had to wholly disable notifications to make it stop. Don't really want to have to leave notifications wholly-disabled within the app (though an in-app, easy to find DND toggle would also be helpfu), but I also need to not worry if late-night messages are going to wake me up.

**Comment by kakaroto** [comment]
I don't know if it's a regression or this was never fixed, but I got a rude awakening this morning as well when a Signal call rang despite the phone being in do not disturb mode.
Phone is a One Plus 12 running Android 15 and Signal version 7.28.4

**Comment by ferricoxide** [comment]
Ugh… Just noticed that this ticket, like the others that are either mentioned in it or similarly-report the issue have either been explicitly-marked "closed" or have been marked as "stale" by a bot, yet the problem persists. 

Can someone on the Signal team either have the decency to say "we're not going to fix this: pound sand" or actually triage the ticket and provide some sort of estimate on when it will be fixed?

**Comment by githubuser6000** [comment]
I cancelled my donations, including matching. I was planning on trying to go through the codebase, but given their response to [this](https://github.com/signalapp/Signal-Desktop/pull/6186), I'd rather spend it elsewhere.

**Comment by CommonLoon102** [comment]
> I cancelled my donations, including matching. I was planning on trying to go through the codebase, but given their response to [this](https://github.com/signalapp/Signal-Desktop/pull/6186), I'd rather spend it elsewhere.

They gave proper feedback and listed the important details what the PR was missing. Also, I can understand if there are more important things than GIFs. Probably it is the least important feature in Signal. I don't know why they have even implemented it the first time.

**Comment by ferricoxide** [comment]
> > I cancelled my donations, including matching. I was planning on trying to go through the codebase, but given their response to [this](https://github.com/signalapp/Signal-Desktop/pull/6186), I'd rather spend it elsewhere.
> 
> They gave proper feedback and listed the important details what the PR was missing. Also, I can understand if there are more important things than GIFs. Probably it is the least important feature in Signal. I don't know why they have even implemented it the first time.

GIFs??? Nah, dude, this ticket's about the fact that Signal doesn't honor DND and making a right nuisance of itself. Giving people the choice between getting an uninterrupted night's sleep or not getting notifications _at all_? That's pretty effing problematic.

**Comment by ferricoxide** [comment]
> > I cancelled my donations, including matching. I was planning on trying to go through the codebase, but given their response to [this](https://github.com/signalapp/Signal-Desktop/pull/6186), I'd rather spend it elsewhere.
> 
> They gave proper feedback and listed the important details what the PR was missing. Also, I can understand if there are more important things than GIFs. Probably it is the least important feature in Signal. I don't know why they have even implemented it the first time.

While the issue that the `this` links to would rightly be low-priority, I'd wager that it's a "last straw" kind of thing. I mean, not honoring DND is pretty freaking problematic and they've had a few tickets about it that they've closed without fixing. Giving people the choice between getting an uninterrupted night's sleep or not getting notifications _at all_? That's pretty freaking problematic and a great way to kill people's interest in a service.

**Comment by githubuser6000** [comment]
@CommonLoon102 

I've seen the same arguments made in the community forums and reminds me of gaslighting. But isn't that forum so I'm trying to give you benefit of the doubt here. 

> They gave proper feedback and listed the important details what the PR was missing. 

No, not really. If you read the thread more carefully, you'll notice multiple questions were left unanswered. 

> Also, I can understand if there are more important things than GIFs. Probably it is the least important feature in Signal. 

Sure, that's a fine argument to make. But then don't get sad/mad when people stop or don't want to use signal. I've gotten thinly-veiled insults in the community forum because of contacts leaving while I was sharing the issues and concerns they had .... The "plain" version (with the said insult-y part removed) is "friends don't let friends use <insert app with bad privacy>". 

"But privacy and security!", I hear you cry? Well that's true too. 

In the world of other other messengers, why would the common joe move to signal? 

Why are other messengers so popular? Because *they do what the user wants*. Unless you mean to say signal isn't for the common joe. That's fine too. If that's the case, then Signal needs to make it clear "Not for the common user" so people don't come in with the wrong expectations. 

> I don't know why they have even implemented it the first time.

Assuming that isn't a rhetorical question, because people asked for it? Because everyone else has it. Because otherwise people wouldn't feel motivated to switch? Signal usernames were "in the pipe" for sooooo long and were "prioritized" only after whatsapp released them. Signal used to lead the way, and of late, it follows, dare I say, limps behind.

And I'll leave it at that to prevent off-topic discussion (for this thread). 

---
ferricoxide:

> I'd wager that it's a "last straw" kind of thing.

Nailed it. 

> they've had a few tickets about it that they've closed without fixing

Not to mention changing things and *THEN* telling users about it. I am still salty about drafts disappearing, deleted messages being deleted across all devices without a way to opt-out.

**Comment by greyson-signal** [comment]
Hey folks, I can see there's a lot of frustration. For what it's worth, this isn't necessarily a widespread issue, but rather an issue that's limited to certain devices or configurations. We have people on the team that use and rely on DnD, and we've tested several of our test devices, and unfortunately cannot reproduce the issue.

The next step for us is likely to buy devices that are affected by the issue. For people that are affected, can you please post a link to a [debuglog](https://support.signal.org/hc/en-us/articles/360007318591-Debug-Logs-and-Crash-Reports)? Also, can you share what relevant DnD settings you have enabled, like whether repeat callers are enabled?

Thanks!

**Comment by kakaroto** [comment]
> can you share what relevant DnD settings you have enabled, like whether repeat callers are enabled?

I have repeat callers enabled, but also, I have exceptions for calls and messages set to "Favorite contacts". I hope that helps.
I'll see about getting you a debuglog.

**Comment by DanielBarie** [comment]
DND enabled, repeat callers enabled.
DND specific: 
- Some callers may interrupt (repeat yes, but no messages)
- Apps may interrupt: Digital Wellbeing, Google Play
- Alarms: Alarms yes, media yes
- Options for filtered: No acoustic signal but shown

I don't have a debug log at hand for when it happened. Will any log do?

**Comment by marcinsmialek** [comment]
New interesting finding here:  
I've got Samsung S23U, Android 14, S918BXXU7CXK4

When I have DND enabled, and the contact is in the DND exceptions list, then I get notification. But now I can see the same behavior from both Signal and Telegram!  
They make my phone vibrate or not in the same conditions.

Below there are DND exceptions settings and effects when receiving a message. DND enabled for all scenarios, and the contact is marked as favorite. Both for Signal and Telegram:  

| Calls from ⭐ | Messages from ⭐ | Contact in exceptions | Notification |
|---|---|---|---|
| Disabled | Disabled | No | Silent |
| Enabled | Disabled | No | **Silent** 😸 |
| Disabled | Enabled | No | Vibrates 😿 |
| Any | **Enabled** | Any | Vibrates 😿 |
| Disabled | Disabled | Yes | Vibrates 😿 |
| Any | Any | **Yes** | Vibrates 😿 |

It appears that now it's the way to decide how to receive notifications.  
Changing Signal notifications profile to silent does the job too. It would be cool if we could sync it with Android DND mode natively, or with Tasker.

**Comment by ElijahLynn** [comment]
> The next step for us is likely to buy devices that are affected by the issue. For people that are affected, can you please post a link to a [debuglog](https://support.signal.org/hc/en-us/articles/360007318591-Debug-Logs-and-Crash-Reports)? Also, can you share what relevant DnD settings you have enabled, like whether repeat callers are enabled?

So, you have a bunch of users that can replicate the issue. As a software engineer myself, what I would do is jump on a call with some of these users and get realtime debug feedback. The feedback loop is too slow like this, it needs to be faster.

**Comment by mtang-signal** [comment]
>  Only `contacts` exempted are my Favorites – but was under the impression that that only effected phone calls. 

@ferricoxide Thanks for the log! Just to clarify - your Favorites being exempted from DND also applies to Signal calls and not just phone calls. So if your DND setting for calls is set to Favorites, and then they call you via Signal, we'll send a notification. Same logic applies for messages. Could you just confirm that there are non-favorited contacts that will cause a notification even while you're on DND?

**Comment by ferricoxide** [comment]
> [@ferricoxide](https://github.com/ferricoxide) Thanks for the log! Just to clarify - your Favorites being exempted from DND also applies to Signal calls and not just phone calls. So if your DND setting for calls is set to Favorites, and then they call you via Signal, we'll send a notification. Same logic applies for messages. Could you just confirm that there are non-favorited contacts that will cause a notification even while you're on DND?

Ok. Weird. Hadn't anticipated that, since I've mostly added Signal contacts since Signal got rid of "real phone number" contacts-management. Assumed that my Signal app wouldn't be able to map from my phone's native contacts-manager given the (seeming) deprecation of use of real phone numbers for contact-management?

**Comment by mtang-signal** [comment]
> When I have DND enabled, and the contact is in the DND exceptions list, then I get notification.

@marcinsmialek Thanks for doing so much investigation here. If a contact is in exceptions, they'll override the Calls/Messages DND settings and you'll get a notification. So the last three lines in the table is correct behavior.

This line here is concerning:

> Calls from ⭐ | Messages from ⭐ | Contact in exceptions | Notification
> -- | -- | -- | --
> Disabled | Enabled | No | Vibrates 😿

Would you be able to provide more details on this? Specifically, was the notification that triggered the vibration from a favorited contact? And was the notification a call or a message? If it was a favorite contact, then only message notifications should be going through.

**Comment by mtang-signal** [comment]
> I have repeat callers enabled, but also, I have exceptions for calls and messages set to "Favorite contacts". I hope that helps. I'll see about getting you a debuglog.

@kakaroto Thanks for the information. Would you happen to remember whether or not the person who called you was a Favorite contact and/or if they called twice in a span of 15 minutes (our qualification for a repeat caller)?

**Comment by kakaroto** [comment]
> 
> [<img alt="" width="16" height="16" src="https://avatars.githubusercontent.com/u/27990?s=64&amp;v=4">@kakaroto](https://github.com/kakaroto?rgh-link-date=2025-01-14T20%3A41%3A26.000Z) Thanks for the information. Would you happen to remember whether or not the person who called you was a Favorite contact and/or if they called twice in a span of 15 minutes (our qualification for a repeat caller)?

It definitely was not a favorite caller as all my contacts on Signal are direct signal users added via username, not associated with any phone contacts.
Also, the call was a single call, not a repeat caller.
I hope that helps narrow it down!
Also note in my case that it rang, it didn't just vibrate, if that makes a difference to you.
Thanks!

**Comment by marcinsmialek** [comment]
> > When I have DND enabled, and the contact is in the DND exceptions list, then I get notification.
> 
> [@marcinsmialek](https://github.com/marcinsmialek) Thanks for doing so much investigation here. If a contact is in exceptions, they'll override the Calls/Messages DND settings and you'll get a notification. So the last three lines in the table is correct behavior.
> 
> This line here is concerning:
> 
> > Calls from ⭐
> > Messages from ⭐
> > Contact in exceptions
> > Notification
> > 
> > 
> > 
> > 
> > Disabled
> > Enabled
> > No
> > Vibrates 😿
> 
> Would you be able to provide more details on this? Specifically, was the notification that triggered the vibration from a favorited contact? And was the notification a call or a message? If it was a favorite contact, then only message notifications should be going through.

Yeah, it was a text message from a fav contact.

One one hand it seems like "by design", but on the other hand, it's troublesome, especially for calls.  

My use case is that I don't use text messages nor call from apps with DND enabled. DND meant no contact except for truly emergency communications.  
Because I can't set per contact exceptions separately for calls and messages, I have to manage the exceptions via favorites list, which grows and becomes cumbersome to use.

Signal supports notification profiles. Could we get a profile which would be DND-aware? Like 2 toggles for blocking messages and calls when DND is on?

**Comment by greyson-signal** [comment]
Hey folks, we believe we've found one bug where we may incorrectly ignore DnD if the "allow repeat callers" setting is enabled. Fix for that going in 7.30. We also added a bunch more logging to aid in tracking down any other remaining issues. Thanks!

**Comment by ElijahLynn** [comment]
That is promising! I don't have "allow repeat callers" on right now, but I swear I did. I'll do some more testing with that setting. 

Also, fwiw, current Android version seems to be 7.28.4 for those of us using the Play Store.

**Comment by ferricoxide** [comment]
Ok, so, I tried creating/using Notification Profiles. I created two: a "weekdays"  and a "weekends" profile to try mimic my device-level automatic DnD. I set up each with no contacts/groups in the "Allowed notifications" list.

First bit of weirdness is it seems one can only have one active profile?
Second bit of weirness is that the little "sleep" moon shows up within the app, no matter whether I'm  within the configured "no text notifications" time-window or not.  No matter the time of day, I seem to have not heard my phone's Signal app chirp since then to alert me of new texts.

So, I'm guessing I've fundamentally misunderstood how to use profiles to work around the Signal app's "we'll scream about new texts from your favorites, no matter the time of day" problem?

**Comment by Ablesius** [comment]
Hi, adding FairPhone 4 to the list of affected devices!

> We have been unable to reproduce this issue on our end. A few clarifying questions:
> 
>     * Is it all conversations or just some?

Just some. I've changed my DND setting to not even show notifications on the lockscreen, and some got through; when I turned it off, I could see that others had successfully been blocked. "Some" that got through were both from starred contacts (in Android) and from not-starred. For me it's unclear which chats will go through with notifications and which won't - I thought _maybe_ "users I have most recently chatted with get through" but I can't confirm that.


>     * Do you have the contacts in question as "starred" or "priority" contacts?

Some, see above.


>     * Do you have the conversations as "priority conversations" in the system settings?

No.


>     * If you have other devices (or friends with other Android devices), can you confirm if it reproduces on theirs? (and if so, please share the manufacturer and device model)

Model is "FP4", do you need serial number and build number or something?

**Comment by film4theblind-rgb** [comment]
Started happening again with version 7.58.2.

**Comment by kakaroto** [comment]
> Started happening again with version 7.58.2.

Can confirm. I think it started happening for me 2 weeks ago though (so before the 7.58.2 release? I'm not 100% sure on the timeline of noticing the issue).
In this case, it was text messages from a specific user (the user is a favorite in my phone) and the notifications woke me up for a few days until I muted the chat with the friend.

**Comment by AniAggarwal** [comment]
Can confirm this occurred for me before the 7.58.2 release, on a 7.57.x version (Samsung S23 Ultra)
I experienced two separate behaviors:
1. Same as others, heard ringtone and screen changed to show call from a favorited caller's first call while on DND.
2. In a group chat with only non-favorite contacts, I did not hear ringing. However, when the call either finished among the group or ringing finished, I heard the Signal 'hang-up' sound loudly. This is the sound you normally hear you you or someone else hangs up the call that you are both in, but I was not in the call and the phone was not on.

**Comment by clavelm** [comment]
I'm on One Plus 10 pro (EU) with Oxygen OS NE2213_15.0.0.1301 and Signal 7.62.3.
I has been having the problem with previous version of the OS and Signal, it's not a recent thing.

My phone is nearly always on DND but my notification configuration for Signal is to allow notifications while on DND.

Whenever I deactivate and reactivate DND, Signal does not emit notification sounds anymore.

The switch to allow notifications on DND is still set, and I'm still getting banners for notifications.
Workaround to get back to working notification sounds is to deactivate and reactivate the "allow while on DND" switch.

Whatsapp has the same settings on my phone and does not have that problem so I'm pretty sure it's coming from Signal.
